### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by carolinafm216**

### CODE REVIEW

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]
You should have used the eurostat_life_expect.json which was in https://github.com/nosportugal/faast-foundations/blob/main/assignments/assignment_4/eurostat_life_expect.zip to test your json strategy

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]
You should have used the eurostat_life_expect.json which was in https://github.com/nosportugal/faast-foundations/blob/main/assignments/assignment_4/eurostat_life_expect.zip to test your json strategy

File: assignments/life_expectancy/tests/pyproject.toml [hunk 1]
```diff
@@ -30,4 +30,5 @@ disable = [
    "missing-function-docstring",
    "line-too-long",
    "invalid-name",
+   "bad-classmethod-argument",
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/pyproject.toml:33]: ok

File: assignments/life_expectancy/Country.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from enum import Enum
+
+import pandas as pd
+
+
+# mudar nome do ficheiro para country - pap8
+class Country(Enum):
+    ES = "ES"
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/Country.py:8]: ok

File: assignments/life_expectancy/tests/test_load_save_data_new.py [hunk 3]
```diff
@@ -13,15 +14,15 @@ def test_load_data(life_expectancy_filter_region):
     with patch("life_expectancy.loading_saving_data.pd.read_csv") as mock_read_csv:
         mock_read_csv.return_value = life_expectancy_filter_region  # imput fixture
 
-        raw_actual = load_data(file_path)
+        File_processor = file_processor(file_path, file_processor.CSVReadingStrategy())
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_load_save_data_new.py:17]: ok

File: assignments/life_expectancy/pyproject.toml [hunk 4]
```diff
@@ -27,6 +27,8 @@ fail_under = 80
 [tool.pylint."messages control"]
 disable = [
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/pyproject.toml:28]: ok

File: assignments/life_expectancy/loading_saving_data.py [hunk 5]
```diff
@@ -1,17 +1,10 @@
 import pathlib
 import pandas as pd
 
-read_file_name = "eu_life_expectancy_raw.tsv"
+read_file_name = "life_expectancy_original.json"
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/loading_saving_data.py:4]: ok

File: assignments/life_expectancy/loading_saving_data.py [hunk 6]
```diff
@@ -1,17 +1,10 @@
...
 file_path = pathlib.Path(__file__).parent / "data" / read_file_name
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/loading_saving_data.py:5]: ok

File: assignments/life_expectancy/__init__.py [hunk 7]
```diff
@@ -1,10 +1,29 @@
...
+from life_expectancy.loading_saving_data import save_data
+
 
 if __name__ == "__main__":  # pragma: no cover
-    save_data(clean_data(load_data(LIFE_EXPECTANCY_FILE_PATH), "PT"))
+    file = "eu_life_expectancy_raw.tsv"
+    file_path = pathlib.Path(__file__).parent / "data" / file
+
+    if file.split(".")[1] == "json":
+        File_processor = file_processor(file_path, JSONReadingStrategy())
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/__init__.py:19]: ok

**Comment by carolinafm216** [comment]
feito!

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]
Still have some unresolved comments

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]
Still have some unresolved comments

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]
Still have some unresolved comments

File: assignments/life_expectancy/Country.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/Country.py:22]: feito

File: assignments/life_expectancy/tests/test_load_save_data_new.py [hunk 8]
```diff
@@ -13,15 +14,16 @@ def test_load_data(life_expectancy_filter_region):
     with patch("life_expectancy.loading_saving_data.pd.read_csv") as mock_read_csv:
         mock_read_csv.return_value = life_expectancy_filter_region  # imput fixture
 
-        raw_actual = load_data(file_path)
+        File_Processor = file_processor(file_path, file_processor.CSVReadingStrategy())
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_load_save_data_new.py:17]: Feito

File: assignments/life_expectancy/tests/test_load_save_data_new.py [hunk 9]
```diff
@@ -13,15 +14,16 @@ def test_load_data(life_expectancy_filter_region):
...
+        Raw_Actual = File_Processor.load_data()
 
-        pd.testing.assert_frame_equal(raw_actual, life_expectancy_filter_region)
+        pd.testing.assert_frame_equal(Raw_Actual, life_expectancy_filter_region)
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_load_save_data_new.py:20]: ok

File: assignments/life_expectancy/tests/test_country.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    life_expectancy_unpivot,
+)
+
+read_file_name = "mock_file_path.tsv"
+file_path = pathlib.Path(__file__).parent / "data" / read_file_name
+
+
+def test_list_country(coutry_list_expected, life_expectancy_unpivot):
+    response = Country.list_all_countries(life_expectancy_unpivot)
+    assert response == coutry_list_expected
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_country.py:10]: I don't get what do you mean by this comment

**Review by margarida-borbinha — APPROVED** [review]

**Comment by carolinafm216** [comment]
Assigment 4

**Pull request merged**